### PR TITLE
Add reconciler default timout, cluster and machine scopes, machine `Reconcile` function, and `InstanceStatus` enums

### PR DESCRIPTION
**Pull request by zliang-akamai**
Some basic components

### CODE REVIEW

File: cloud/scope/cluster.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+// ClusterScope defines the basic context for an actuator to operate upon.
+type ClusterScope struct {
+	client      client.Client
+	patchHelper *patch.Helper
+
+	LinodeClients
+	Cluster    *clusterv1.Cluster
+	GCPCluster *infrav1.LinodeCluster
```
Comment (luthermonson) [inline cloud/scope/cluster.go:50]: gcp cluster eh?

File: cloud/scope/clients.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+limitations under the License.
+*/
+
+package scope
+
+import (
+	"github.com/linode/linodego"
+)
+
+type LinodeClients struct {
```
Comment (luthermonson) [inline cloud/scope/clients.go:23]: i might be missing context on other usages of the linodego client... is this plural on purpose? At present I don't see the benefit over doing it this way than just adding a `lindoeClient` next to the k8s client on the on the scope.

Comment (zliang-akamai) [reply cloud/scope/clients.go]: I stole the pattern from the achieved provider and other provider and but later realized we just have one client. Thanks for the comment!

Comment (zliang-akamai) [reply cloud/scope/clients.go]: After digging into the cluster API provider implementation, now I think the `LinodeClients` struct is still useful for centralizing Linode client lifecycle management logic, e.g. client creation. 

[This was found in the achieved version of the CAPI provider.](https://github.com/linode/cluster-api-provider-linode-archive/blob/master/cloud/scope/sessions.go#L30-L45) But yea, it is not that necessary and we should be fine for now with the a private function to create the client in cluster scope package. We can always move to this pattern later when we find it's more useful/necessary (unless you think we should do it now, lol).

**Review by luthermonson — APPROVED** [review]
great!

**Pull request merged**